elastic/beats
# Updating elasticsearch/ccr metricset with ES ccr_stats API changes

Resolves https://github.com/elastic/kibana/issues/25778.

The ES CCR API changed a little before the 6.5.0 FF and we didn't catch these changes 😞. This PR updates the CCR metricset code to work with the new ES CCR API response structure.

## Testing this PR
1. Setup CCR: https://www.elastic.co/guide/en/elastic-stack-overview/current/ccr-getting-started.html
2. Checkout this PR and `make` Metricbeat: `cd metricbeat; make`
3. Enable the `elasticsearch` module: `./metricbeat modules enable elasticsearch`
4. Edit `modules.d/elasticsearch.yml` and make sure the `ccr` metricset is enabled and `xpack.enabled: true` is set.
5. Start Metricbeat: `./metricbeat -e`
6. After about 10-20 seconds, check that Metricbeat indexed the correct document of `type` = `ccr_stats` into `.monitoring-es-mb-6-*`.
